pyiceberg/expressions/__init__.py

```diff
@@ -559,8 +561,9 @@ def as_bound(self) -> Type[BoundNotNaN[L]]:
         return BoundNotNaN[L]
 
 
-class SetPredicate(UnboundPredicate[L], ABC):
-    literals: Set[Literal[L]]
+class SetPredicate(UnboundPredicate[L], IcebergBaseModel, ABC):
```

Comment:
I think this should supress the warning for now:
```suggestion
class SetPredicate(UnboundPredicate[L], IcebergBaseModel, ABC):
    model_config = ConfigDict(arbitrary_types_allowed=True)

```

In the end, the whole  expression tree should be serializeable to JSON, and then we can remove this configuration again.